packages/block-editor/src/components/block-tools/index.js

```diff
@@ -64,6 +64,55 @@ export default function BlockTools( {
 		moveBlocksDown,
 	} = useDispatch( blockEditorStore );
 
+	const isLargerThanTabletViewport = useViewportMatch( 'medium', '>' );
+
+	useLayoutEffect( () => {
+		// don't do anything if not fixed toolbar
+		if ( ! hasFixedToolbar || ! isLargerThanTabletViewport ) {
+			return;
+		}
+
+		const blockToolbar = document.querySelector(
```

Comment:
This might be a bit far-fetched but could we instead of querying the DOM check for the [same conditions ](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-tools/block-contextual-toolbar.js#L80)that allow the contextual toolbar to render? And then pass the styles to the contextual toolbar component instead of setting them directly below?

Otherwise, can we reuse this `blockToolbar` const below instead of repeating the query?

Reply:
We need to know if the toolbar is visible here so we can apply the styles to it depending on the width of all the pinned items. From settings we can know the toolbar should show, but we don't know if it is actually in the dom, and then we get a problem when trying to set the dynamic width.

I will reuse the variable though 🙏🏻